File: backend-spring-boot/src/main/java/com/errormonitoring/model/ErrorLog.java
package com.errormonitoring.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "error_logs", indexes = {
    @Index(name = "idx_application_id", columnList = "application_id"),
    @Index(name = "idx_severity", columnList = "severity"),
    @Index(name = "idx_created_at", columnList = "created_at")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class ErrorLog {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    
    @Column(name = "application_id", nullable = true)
    private Integer applicationId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "application_id", insertable = false, updatable = false)
    private Application application;
    
    @Column(nullable = false, length = 2000, columnDefinition = "TEXT")
    private String message;
    
    @Column(name = "stack_trace", length = 5000, columnDefinition = "TEXT")
    private String stackTrace;
    
    @Column(length = 200)
    private String source;
    
    @Column(nullable = false, length = 20)
    @Enumerated(EnumType.STRING)
    private ErrorSeverity severity;
    
    @Column(length = 50)
    private String status = "Open";
    
    @Column(length = 50)
    private String environment;
    
    @Column(name = "user_agent", length = 500)
    private String userAgent;
    
    @Column(name = "ip_address", length = 45)
    private String ipAddress;
    
    @Column(name = "http_method", length = 10)
    private String httpMethod;
    
    @Column(name = "api_endpoint", length = 500)
    private String apiEndpoint;
    
    @Column(name = "error_type", length = 100)
    private String errorType;
    
    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @Column(name = "resolved_at")
    private LocalDateTime resolvedAt;
    
    public enum ErrorSeverity {
        Low, Medium, High, Critical
    }
}

File: backend-error/Models/ErrorLog.cs
using System.ComponentModel.DataAnnotations;

namespace ErrorMonitoringAPI.Models;

public class ErrorLog
{
    public int Id { get; set; }
    
    [Required]
    [Range(1, int.MaxValue, ErrorMessage = "ApplicationId must be a valid application ID")]
    public int ApplicationId { get; set; }
    
    [Required(ErrorMessage = "Error message is required")]
    [StringLength(2000, ErrorMessage = "Error message cannot exceed 2000 characters")]
    public string Message { get; set; } = string.Empty;
    
    [StringLength(5000, ErrorMessage = "Stack trace cannot exceed 5000 characters")]
    public string? StackTrace { get; set; }
    
    [StringLength(200, ErrorMessage = "Source cannot exceed 200 characters")]
    public string? Source { get; set; }
    
    [Required(ErrorMessage = "Severity is required")]
    [RegularExpression("^(Low|Medium|High|Critical)$", ErrorMessage = "Severity must be Low, Medium, High, or Critical")]
    public string Severity { get; set; } = string.Empty;
    
    [StringLength(50, ErrorMessage = "Status cannot exceed 50 characters")]
    public string? Status { get; set; } = "Open";
    
    [StringLength(50, ErrorMessage = "Environment cannot exceed 50 characters")]
    public string? Environment { get; set; }
    
    [StringLength(500, ErrorMessage = "User agent cannot exceed 500 characters")]
    public string? UserAgent { get; set; }
    
    [StringLength(45, ErrorMessage = "IP address cannot exceed 45 characters")]
    public string? IpAddress { get; set; }
    
    [StringLength(10, ErrorMessage = "HTTP method cannot exceed 10 characters")]
    public string? HttpMethod { get; set; }
    
    [StringLength(500, ErrorMessage = "API endpoint cannot exceed 500 characters")]
    public string? ApiEndpoint { get; set; }
    
    [StringLength(100, ErrorMessage = "Error type cannot exceed 100 characters")]
    public string? ErrorType { get; set; }
    
    [Required]
    public DateTime CreatedAt { get; set; }
    
    public DateTime? ResolvedAt { get; set; }
    
    // Navigation property
    public Application? Application { get; set; }
}